esp-hal/src/uart.rs

```diff
@@ -49,119 +36,47 @@
 //! available. See the examples below for more information on how to interact
 //! with this driver.
 //!
-//! ## Examples
-//! ### Sending and Receiving Data
-//! ```rust, no_run
-#![doc = crate::before_snippet!()]
-//! # use esp_hal::uart::{Config, Uart};
-//! # let mut uart1 = Uart::new(
-//! #     peripherals.UART1,
-//! #     Config::default(),
-//! # ).unwrap()
-//! # .with_rx(peripherals.GPIO1)
-//! # .with_tx(peripherals.GPIO2);
-//! // Write bytes out over the UART:
-//! uart1.write_bytes(b"Hello, world!").expect("write error!");
-//! # }
-//! ```
-//! 
-//! ### Splitting the UART into RX and TX Components
-//! ```rust, no_run
-#![doc = crate::before_snippet!()]
-//! # use esp_hal::uart::{Config, Uart};
-//! # let mut uart1 = Uart::new(
-//! #     peripherals.UART1,
-//! #     Config::default(),
-//! # ).unwrap()
-//! # .with_rx(peripherals.GPIO1)
-//! # .with_tx(peripherals.GPIO2);
-//! // The UART can be split into separate Transmit and Receive components:
-//! let (mut rx, mut tx) = uart1.split();
-//!
-//! // Each component can be used individually to interact with the UART:
-//! tx.write_bytes(&[42u8]).expect("write error!");
-//! let mut byte = [0u8; 1];
-//! rx.read_bytes(&mut byte);
-//! # }
-//! ```
-//! 
-//! ### Inverting RX and TX Pins
-//! ```rust, no_run
-#![doc = crate::before_snippet!()]
-//! # use esp_hal::uart::{Config, Uart};
-//!
-//! let (rx, _) = peripherals.GPIO2.split();
-//! let (_, tx) = peripherals.GPIO1.split();
-//! let mut uart1 = Uart::new(
-//!     peripherals.UART1,
-//!     Config::default())
-//!     .unwrap()
-//!     .with_rx(rx.inverted())
-//!     .with_tx(tx.inverted());
-//! # }
-//! ```
-//! 
-//! ### Constructing RX and TX Components
-//! ```rust, no_run
-#![doc = crate::before_snippet!()]
-//! # use esp_hal::uart::{Config, UartTx, UartRx};
+//! ## Example
 //!
-//! let tx = UartTx::new(
-//!     peripherals.UART0,
-//!     Config::default())
-//!     .unwrap()
-//!     .with_tx(peripherals.GPIO1);
-//! let rx = UartRx::new(
-//!     peripherals.UART1,
-//!     Config::default())
-//!     .unwrap()
-//!     .with_rx(peripherals.GPIO2);
-//! # }
-//! ```
-//! 
-//! ### Operation with interrupts that by UART/Serial
+//! ### Handling UART/Serial Interrupts
 //! Notice, that in practice a proper serial terminal should be used
 //! to connect to the board (espmonitor and espflash won't work)
 //! ```rust, no_run
 #![doc = crate::before_snippet!()]
 //! # use esp_hal::delay::Delay;
 //! # use esp_hal::uart::{AtCmdConfig, Config, Uart, UartInterrupt};
-//! let delay = Delay::new();
-//!
-//! // Default pins for UART/Serial communication
+//! # let delay = Delay::new();
 #![cfg_attr(
     esp32,
-    doc = "let (tx_pin, rx_pin) = (peripherals.GPIO1, peripherals.GPIO3);"
+    doc = "# let (tx_pin, rx_pin) = (peripherals.GPIO1, peripherals.GPIO3);"
```

Comment:
If you're hiding these, let's remove all the cfg_attrs and just assign a `NoPin` to both variables.